### ISSUE TITLE
Rewrite it in rust?

### ISSUE THREAD

**Issue by agauniyal** [avinassh/gg-flip#17]
Writing [Rust](https://www.rust-lang.org/) is like going to a monastery to practice the art of **amazing programming**. The initial curve can be brutal: you will get beaten by _borrow checker_ for **unsafe habits**, you will have to recite _parametric trait mantras_ every day and contemplate the Zen of lifetime and _thread-safety_. It will be painful, but it's a **path to enlightenment**. Even when you leave the monetary, you will be another person. You will start to appreciate safe methods, right ways of sharing data, _unbreakable APIs_.

In addition to that, rust also has :zero: **zero-cost abstractions** :zero:, 🏃**move semantics** 🏃, 💯 **guaranteed memory safety** 💯, 🏁 **threads without data races** 🏁, 🌪 **trait-based generics** 🌪, 🔄 **pattern matching** 🔄, 🔩 **type inference** 🔩, 🙅‍♂️ **no runtime** 🙅‍♂️ and 🔥 **efficient C bindings** 🔥.

PS - did I mention its ⚡️ **_webscale_** ⚡️?

**Comment by avinassh** [comment]
I agree with all the above, however, Rust has generics which really makes that as an unsuitable candidate for GG Flip as compared to Go. However, I will still keep this thread open for more discussions and ideas. Perhaps if someone forks Rust and compiles it without generics.

**Comment by BradfordMedeiros** [comment]
I have HUGE ISSUES with this.
RUST is safe! Are we really willing to invoke the garbage collector?  We need to remember to keep it SOLID

Besides I'm not sure if GO is really webscale like you claim it is.

**Comment by BradfordMedeiros** [comment]
Also we should consider adding mongo or maybe be traditional and postgres/mysql/mariadb to be able to support more sign flip types.  Looks like we have an enumerated list of sign flip capabilities, but we need to be able to support every possible flip.
